cmd/notary/tuf.go

```diff
@@ -270,58 +272,82 @@ func verify(cmd *cobra.Command, args []string) {
 	return
 }
 
-func passphraseRetriever() (string, error) {
-	fmt.Println("Please provide a passphrase for this root key: ")
-	var passphrase string
-	_, err := fmt.Scanln(&passphrase)
-	if err != nil {
-		return "", err
-	}
-	if len(passphrase) < 8 {
-		fmt.Println("Please use a password manager to generate and store a good random passphrase.")
-		return "", errors.New("Passphrase too short")
-	}
-	return passphrase, nil
-}
+func getNotaryPassphraseRetriever() (trustmanager.PassphraseRetriever) {
+	userEnteredTargetsSnapshotsPass := false
+	targetsSnapshotsPass := ""
 
-func getPassphrase(confirm bool) ([]byte, error) {
-	if pass := os.Getenv("NOTARY_ROOT_PASSPHRASE"); pass != "" {
-		return []byte(pass), nil
-	}
+	return func(keyID string, alias string, createNew bool, numAttempts int) (string, bool, error) {
```

Comment:
You caught me, I was being clever. There is a slight advantage to this model in that I can eventually use this method for the import/export flow where I create a closure for each of the keystores (one for import, one for export). 

I'm not sure how much the synchronization issues will affect us, as in the worst case the code will ask twice, but I don't imagine code with goroutines will be prompting the user multiple times. Even so, I think a synchronized version of passphraseRetriever could be implemented for that use case.